src/extras.py

```diff
@@ -46,19 +55,20 @@ def augment_with_salary(record: Record) -> str:
 ########################################################################################
 # OpenOversight ID mapping
 ########################################################################################
-def _load_oo_id_mapping() -> None:
-    with OO_MAPPING_PATH.open(newline="") as csvfile:
-        reader = csv.DictReader(csvfile)
-        # Do this rather than a dict comprehension so we populate the global object
+def _get_oo_id_mapping() -> Mapping[str, str]:
+    if OO_CACHE.currsize == 0:
+        log.info("Refreshing OO ID cache")
+        response = requests.get(OO_ID_SOURCE_URL)
+        reader = csv.DictReader(io.StringIO(response.text, newline=""))
         for row in reader:
-            OO_ID_MAPPING[row["badge number"]] = row["id"]
+            OO_CACHE[row["badge number"]] = row["id"]
+    return OO_CACHE
 
 
 def augment_with_oo_link(record: Record) -> str:
-    if not OO_ID_MAPPING:
-        _load_oo_id_mapping()
-    if oo_id := OO_ID_MAPPING.get(record["badge"]):  # type: ignore
-        oo_link = OO_URL.format(id_=oo_id)
+    id_mapping = _get_oo_id_mapping()
+    if oo_id := id_mapping.get(record["badge"]):  # type: ignore
+        oo_link = OO_URL_TEMPLATE.format(id_=oo_id)
         return render_template("extras/seattle_oo_id.j2", oo_link=oo_link)
     return ""
```

Comment:
Why return an empty string here?

Reply:
This function is expected to return a string, which gets added together with the other augments (line 87 below). If this didn't return an empty string it would just return `None`, which raises a type error when concatenated with a string:

```python
[ins] In [1]: "String " + None
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-332c4aa03a09> in <module>
----> 1 "String " + None

TypeError: can only concatenate str (not "NoneType") to str
```